concordium-std/src/types.rs

```diff
@@ -672,6 +687,56 @@ pub type QueryAccountBalanceResult = Result<AccountBalance, QueryAccountBalanceE
 /// [`QueryContractBalanceError`] and result to [`Amount`].
 pub type QueryContractBalanceResult = Result<Amount, QueryContractBalanceError>;
 
+/// A wrapper around [`Result`] that fixes the error variant to
+/// [`QueryAccountPublicKeysError`] and result to [`AccountPublicKeys`].
+pub type QueryAccountPublicKeysResult = Result<AccountPublicKeys, QueryAccountPublicKeysError>;
+
+/// A wrapper around [`Result`] that fixes the error variant to
+/// [`CheckAccountSignatureError`] and result to [`bool`].
+pub type CheckAccountSignatureResult = Result<bool, CheckAccountSignatureError>;
+
+pub(crate) type KeyIndex = u8;
+
+#[derive(crate::Serialize, Debug, SchemaType)]
+pub(crate) enum PublicKey {
+    Ed25519(PublicKeyEd25519),
+}
+
+#[derive(crate::Serialize, Debug, SchemaType)]
+pub struct CredentialPublicKeys {
+    #[concordium(size_length = 1)]
+    pub(crate) keys:      crate::collections::BTreeMap<KeyIndex, PublicKey>,
```

Comment:
Maybe for another PR in the future, but have you considered using `Vec<(KeyIndex, PublicKey)>` here instead? Since they are never really mutated and probably always a small collection, this might be more efficient depending on the over head of `BTreeMap`... maybe not worth the extra effort.

Reply:
Yes, so that would work if they are only expected to be returned from the node, sicne then we know the order and uniqueness of the keys.

However this type is public, so in principle somebody can pass it as a parameter, in which case we do want to ensure correct and consistent serialization.

Reply:
But it would be more efficient with a Vec, even more efficient would be to not actually do anything to deserialize, just have an iterator, which was my first thought. But then it's hard to use.

Reply:
I have made an issue to address this in the future, if it is important to do so https://github.com/Concordium/concordium-rust-smart-contracts/issues/319